bundles/AdminBundle/Controller/Admin/LoginController.php

```diff
@@ -167,31 +168,46 @@ public function loginCheckAction()
     /**
      * @Route("/login/lostpassword", name="pimcore_admin_login_lostpassword")
      */
-    public function lostpasswordAction(Request $request, ?BruteforceProtectionHandler $bruteforceProtectionHandler, CsrfProtectionHandler $csrfProtection, Config $config, EventDispatcherInterface $eventDispatcher)
+    public function lostpasswordAction(Request $request, ?BruteforceProtectionHandler $bruteforceProtectionHandler, CsrfProtectionHandler $csrfProtection, Config $config, EventDispatcherInterface $eventDispatcher, RateLimiterFactory $resetPasswordLimiter)
     {
         $params = $this->buildLoginPageViewParams($config);
         $error = null;
 
         if ($request->getMethod() === 'POST' && $username = $request->get('username')) {
+
             $user = User::getByName($username);
+            if (!$user instanceof User) {
+                $error = 'user_unknown';
+            }
```

Comment:
Best practice is to not disclose if a reset was unsuccessful because a user account did not exist. This allows enumeration.

Reply:
Should we get rid of the error message completely? or you think keeping other messages like `User is inactive`, `User has no email` etc. still makes sense?

Reply:
Main point is:

> Return a consistent message for both existent and non-existent accounts.

https://cheatsheetseries.owasp.org/cheatsheets/Forgot_Password_Cheat_Sheet.html

Another good read: https://davecallan.com/web-security-account-enumeration-prevention/

Reply:
Thanks! we'll consider it.